### PR TITLE
Added ServiceArea service

### PR DESCRIPTION
**Pull request by vinrran**
Issue #18

### CODE REVIEW

File: src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.sarapis.orservice.controller;
```
Comment (miguel-merlin) [inline src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java:1]: Can you remove the `.DS_Store`. These are binary files autogenerated by your OS.

File: src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/api/service_areas")
+public class ServiceAreaController {
+
+    private final ServiceAreaService serviceAreaService;
+
+    @Autowired
+    public ServiceAreaController(ServiceAreaService serviceAreaService) { //stack overflow the dependency injection paste error and then @autowire
```
Comment (miguel-merlin) [inline src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java:19]: Delete the stack overflow comment

File: src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        this.serviceAreaService = serviceAreaService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ServiceAreaDTO>> getAllServiceAreas() {
+        List<ServiceAreaDTO> serviceAreaDTOs = this.serviceAreaService.getAllServiceAreas();
```
Comment (miguel-merlin) [inline src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java:25]: You don't need to use `this` when calling the `serviceAreaService` as you already set the attribute in the Constructor of the class. Please change it to `serviceAreaService` instead

Comment (miguel-merlin) [reply src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java:25]: Same comment for every other usage of `serviceAreaService`

**Review by miguel-merlin — CHANGES_REQUESTED** [review]

**Comment by vinrran** [comment]
@miguel-merlin I'm not entirely sure, but most likely the error was resolved after I completed the `ServiceAreaServiceImpl.java` file. 

Screenshot 2024-12-12 at 11 48 04 AM

This is a screenshot after I comment out everything in the `ServiceAreaServiceImpl.java` file.

**Comment by miguel-merlin** [comment]
@vinrran Thanks for the clarification!

**Review by miguel-merlin — APPROVED** [review]
lgtm! NIce job!

**Pull request merged**